temporal_sqlalchemy/bases.py

```diff
@@ -9,18 +9,71 @@
 import sqlalchemy as sa
 import sqlalchemy.dialects.postgresql as sap
 import sqlalchemy.orm as orm
+import sqlalchemy.orm.base as base
 import sqlalchemy.orm.attributes as attributes
 import psycopg2.extras as psql_extras
 
 from temporal_sqlalchemy import nine
-from temporal_sqlalchemy.metadata import get_session_metadata
 
-_ClockSet = collections.namedtuple('_ClockSet', ('effective', 'vclock'))
+_PersistentClockPair = collections.namedtuple('_PersistentClockPairs',
+                                              ('effective', 'vclock'))
 
 T_PROPS = typing.TypeVar(
     'T_PROP', orm.RelationshipProperty, orm.ColumnProperty)
 
 
+class ActivityState:
+    def __set__(self, instance, value):
+        assert instance.temporal_options.activity_cls, "Make this better Joey"
+        # TODO should not be able to change activity once changes have been made to temporal properties
+        setattr(instance, '__temporal_current_activity', value)
+
+        if value:
+            current_clock = instance.current_clock
+            current_clock.activity = value
+
+    def __get__(self, instance, owner):
+        if not instance:
+            return None
```

Comment:
You might want to use `return self` here, because otherwise any attempt to access `reset_activity` or `activity_required` from a subclass of `Clocked` will crash with an `AttributeError`:

A minimal example:

```py
import types

ns = types.SimpleNamespace(activity=1)

# This works
ActivityState.reset_activity(ns, None)

class B:
    astate = ActivityState()

# This crashes
B.astate.reset_activity(ns, None)
```